Toolchain/BuildIWYU.sh

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env bash
+set -eo pipefail
+
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+
+ARCH=${ARCH:-"i686"}
+
+BUILD="$DIR/Build/iwyu/$ARCH"
+PREFIX="$DIR/Local/iwyu/$ARCH"
+
+
+IWYU_VERSION=0.16
+IWYU_NAME="include-what-you-use-$IWYU_VERSION"
+IWYU_PKG="$IWYU_NAME.tar.gz"
+IWYU_URL="https://github.com/include-what-you-use/include-what-you-use/archive/refs/tags/$IWYU_VERSION.tar.gz"
+
+
+pushd "$DIR/Tarballs"
+    if [ -e "$IWYU_PKG" ]; then
+        echo "Skipped downloading IWYU"
+    else
+        rm -f "$IWYU_PKG"
+        curl -L "$IWYU_URL" -o "$IWYU_PKG"
+    fi
+
+    if [ -d "$IWYU_NAME" ]; then
+        rm -rf "$IWYU_NAME"
+        rm -rf "$BUILD/iwyu/$ARCH"
+    fi
+
+    echo "Extracting IWYU..."
+    tar -xzf "$IWYU_PKG"
+popd
+
+mkdir -p "$BUILD"
+pushd "$BUILD"
+    cmake "$DIR/Tarballs/$IWYU_NAME" \
```

Comment:
Does using old style invocation serve any purpose here? Serenity requires 3.16 and developers are expected to be running the latest version (which is [trivial](https://alexreinking.com/blog/how-to-use-cmake-without-the-agonizing-pain-part-1.html)) anyway.
```suggestion
cmake -S "$DIR/Tarballs/$IWYU_NAME" -B "$BUILD" \
```
`-S` and `-B` are available since 3.13